### DISCUSSION TITLE
Strelka configuration help

### DISCUSSION THREAD

**Discussion by Mastadamus** [Security-Onion-Solutions/securityonion#3020] [Unsupported Versions]
I have the latest SO build running standalone. I selected Zeek as responsible for metadata. Can someone explain how strelka works in conjunction with this? I've received no alerts from strelka in the main alert tab, however I do show one Strelka file log in Kibana. How exactly does strelka fit in to this stack? I thought that it was supposed to be scanning all files that cross my network at my mirrored port? Or does it only scan files identified as potentially malicious by zeek? How does strelka work exactly and why do i have no alerts, events etc save 1 single log in kibana? i've checked its status using so-status and its up and running. i've also restarted it just to make sure and still nothing. Is absence of logs a good thing and it still working or does that signify its not working right?

**Comment by weslambert** [comment]
Strelka will process the files extracted by Zeek -- if there aren't files being extracted by Zeek, then Strelka will have nothing to scan.  You won't always get Strelka `alerts`, as you shouldn't commonly run across files that would trigger them (YARA rules used from https://github.com/Neo23x0/signature-base).  

Some reasons why you may not being seeing much from Strelka:

- Zeek is dropping packets, or only seeing part of the conversation on the wire, which could affect file extraction (you would need to check the network traffic actually coming in -- sometimes a high capture loss value can indicate this)
- Zeek is not configured to extract the type of files traversing your network
- There are not many files traversing the monitored segment(s)
- Elasticsearch was not able to index the record because of an issue with parsing, etc (check Logstash/Elasticsearch log for clues)

Hope that helps.

Thanks,
Wes

**Reply by Mastadamus** [reply]
Thank you Wes. are you saying that strelka only processes files that are 1.extracted by zeek and 2 match a yara rule? Or does it attempt to process all files extracted by zeek and only "alerts" if it matches a yara rule? the reason why i ask, is i have many files showing in zeke logs, yet only 1 "strelka log" of one processed file by strelka. Its a PDF type and i know i have a bunch of pdf's cross the monitored segment. Additionally, my packet loss for zeek according to grafana is less then 2%  So should it be processing more but only generating alerts if matching a yara rule ? Or did you mean it literally only processes the file if they match a yara rule already?

**Reply by weslambert** [reply]
Strelka processes all files extracted by Zeek. There is only an `alert` generated if that file content matches a YARA rule.

**Comment by innovate-support** [comment]
Have you tried a simulation to see of SO detects it?  It's easier to simulate something rather than wait for real malicious traffic.
https://www.knowbe4.com/ransomware-simulator
https://testmyids.com
